src/overlays/actors/ovl_En_Holl/z_en_holl.c

```diff
@@ -24,37 +36,265 @@ const ActorInit En_Holl_InitVars = {
+        }
+        if (absYDistToPlayer > EN_HOLL_LOADING_PLANE_DISTANCE_VERTICAL) {
+            s32 enHollId = EN_HOLL_GET_ID_CAST(this);
+            s32 playerSide = (this->actor.yDistToPlayer > 0.0f) ? EN_HOLL_ABOVE : EN_HOLL_BELOW;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/EnHoll_Destroy.s")
+            this->actor.room = globalCtx->doorCtx.transitionActorList[enHollId].sides[playerSide].room;
+            if ((this->actor.room != globalCtx->roomCtx.currRoom.num) &&
+                Room_StartRoomTransition(globalCtx, &globalCtx->roomCtx, this->actor.room)) {
+                this->actionFunc = EnHoll_RoomTransitionIdle;
+                this->bgCoverAlphaActive = true;
+            }
+        }
+    } else if (this->bgCoverAlphaActive) {
+        this->bgCoverAlphaActive = false;
+        globalCtx->bgCoverAlpha = 0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_80899ACC.s")
+void EnHoll_VerticalIdle(EnHoll* this, GlobalContext* globalCtx) {
+    f32 absYDistToPlayer;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_80899B88.s")
+    if ((this->actor.xzDistToPlayer < EN_HOLL_RADIUS) &&
+        (absYDistToPlayer = fabsf(this->actor.yDistToPlayer),
+         absYDistToPlayer < EN_HOLL_ACTIVATION_PLANE_DISTANCE_VERTICAL) &&
+        (absYDistToPlayer > EN_HOLL_LOADING_PLANE_DISTANCE_VERTICAL)) {
+        s32 enHollId = EN_HOLL_GET_ID_CAST(this);
+        s32 playerSide = (this->actor.yDistToPlayer > 0.0f) ? EN_HOLL_ABOVE : EN_HOLL_BELOW;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_80899F30.s")
+        this->actor.room = globalCtx->doorCtx.transitionActorList[enHollId].sides[playerSide].room;
+        if ((this->actor.room != globalCtx->roomCtx.currRoom.num) &&
+            Room_StartRoomTransition(globalCtx, &globalCtx->roomCtx, this->actor.room)) {
+            this->actionFunc = EnHoll_RoomTransitionIdle;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_8089A0C0.s")
+void EnHoll_RoomTransitionIdle(EnHoll* this, GlobalContext* globalCtx) {
+    if (globalCtx->roomCtx.unk31 == 0) {
+        func_8012EBF8(globalCtx, &globalCtx->roomCtx);
+        if (globalCtx->bgCoverAlpha == 0) {
+            this->bgCoverAlphaActive = false;
+        }
+        EnHoll_SetupAction(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_8089A238.s")
+void EnHoll_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnHoll* this = THIS;
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_8089A330.s")
+    if ((globalCtx->sceneLoadFlag == 0) && (globalCtx->unk_18B4A == 0) && !(player->stateFlags1 & 0x200)) {
+        this->actionFunc(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/EnHoll_Update.s")
+void EnHoll_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnHoll* this = THIS;
+    Gfx* gfxP;
```

Comment:
Any particular reason to append `P`? Looking through the code base `gfxP` is usually used for arrays of `Gfx *`.

Reply:
Someone in my stream told me that gfxP was the standard name for a variable of type Gfx* (the P stands for the pointer I assume). Would it be better to just call it gfx? I honestly have no idea.

Reply:
I think `dlist` or similar is fine in this case.

Reply:
fixed in https://github.com/zeldaret/mm/pull/270/commits/2209d08d1b50083686d6d560e7fd48c829ff5259 so I'm marking this as resolved.